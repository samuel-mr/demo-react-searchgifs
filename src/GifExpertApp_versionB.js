import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { ListItems_versionB } from './components/ListItems_versionB'

export const GifExpertApp_versionB = ({ defaultArray = ['Do', 'Re', 'Mi'] }) => {

    const [categories, setCategory] = useState(defaultArray)

    return (
        <>
            <h4>GifExpertApp</h4>
            <hr />
            <AddCategory setCategory={setCategory} />
            <ol>
                {
                    categories.map(item => {
                        return <ListItems_versionB key={item} category={item} />
                    })
                }
            </ol>
        </>
    )
}
