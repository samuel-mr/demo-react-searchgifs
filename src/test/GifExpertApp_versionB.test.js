import React from 'react'
import { shallow } from 'enzyme'
import { GifExpertApp_versionB } from '../GifExpertApp_versionB'

describe('GifExpertApp test', () => {
    let wrapper = shallow(<GifExpertApp_versionB />)

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<GifExpertApp_versionB />)
    })

    test('Debe hacer match con el snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe tener un elemento AddCategory', () => {
        const element_addCategory = wrapper.find('AddCategory')
        expect(element_addCategory.exists()).toBeTruthy()
    })

    test('Debe tener una lista de Items', () => {
        const element_list = wrapper.find('ListItems_versionB')
        expect(element_list.exists()).toBeTruthy()
        expect(element_list.length).toBe(3)
    })
    test('Debe tener una lista inicializada con 2 elementos', () => {
        const CATEGORIES = ['gato', 'conejo']
        let wrapper = shallow(<GifExpertApp_versionB defaultArray={CATEGORIES} />)
        const element_list = wrapper.find('ListItems_versionB')
        expect(element_list.exists()).toBeTruthy()
        expect(element_list.length).toBe(CATEGORIES.length)
    })
})