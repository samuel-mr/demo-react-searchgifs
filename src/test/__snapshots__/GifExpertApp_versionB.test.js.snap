// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp test Debe hacer match con el snapshot 1`] = `
<Fragment>
  <h4>
    GifExpertApp
  </h4>
  <hr />
  <AddCategory
    setCategory={[Function]}
  />
  <ol>
    <ListItems_versionB
      category="Do"
      key="Do"
    />
    <ListItems_versionB
      category="Re"
      key="Re"
    />
    <ListItems_versionB
      category="Mi"
      key="Mi"
    />
  </ol>
</Fragment>
`;
